src/System.Data.Odbc/tests/DependencyCheckTest.cs

```diff
@@ -8,7 +8,7 @@ namespace System.Data.Odbc.Tests
 {
     public class DependencyCheckTest
     {
-        [ConditionalFact(Helpers.OdbcNotAvailable)]
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.OdbcNotAvailable), nameof(Helpers.IsNotWindowsServerCore))]
```

Comment:
We should have noticed when we did your last change - I misread the result also - on 32 bit it's not throwing either DllNotFoundException nor OdbcException. 

We don't understand this behavior I assume, so normally we would not just disable the test (unless we leave the issue open)

Since  we have no data on whether System.Data.Odbc even works on Server Core in either bitness I think it's fine to not have an issue, but instead of this change, I would have code like this in the test, to make it clear it's not expected

if (PlatformDetection.IsWindowsServerCore && !Environment.Is64BitProcess)
     return; // On 32 bit Server Core, it does not throw DNFE.